app/controllers/CalculationController.scala

```diff
@@ -66,7 +71,8 @@ class CalculationController @Inject()(viewLPP: CalculationLPPView, viewAdd: Calc
             )
           } else {
             val additionalPenaltyRate = "4"
-            Ok(viewAdd(additionalPenaltyRate))
+            val daysSince31 = ChronoUnit.DAYS.between(penalty.get.financial.dueDate, LocalDateTime.now())
```

Comment:
Since this is additional, the due date (financial.dueDate) should be added to 31,
Calculation = today - (due date + 31 days) - days  
Please correct me if I'm wrong.